Regression: Code does not compile due to undefined symbols

Commit `88bdd434` could be compiled (once the typo in `configure.ac` had been fixed). The current commit `8c80f1c` fails with undefined symbols like `__T0` and `__T1`. I'll try and collect some more detail and update once I have it.

```
/mnt/lustre/tursafs1/home/dp208/dp208/shared/src/grid-sp4-20221213/Grid/qcd/utils/SUn.h: In static member function ‘static void Grid::GaugeGroup<ncolour, group_name>::su2Extract(Grid::Lattice<Grid::iScalar<Grid::iScalar<Grid::iScalar<ttype> > > >&, Grid::Lattice<Grid::iScalar<Grid::iScalar<Grid::iMatrix<vtype, 2> > > >&, const Grid::Lattice<Grid::iScalar<Grid::iScalar<Grid::iMatrix<vtype, ncolour> > > >&, int)’:
/mnt/lustre/tursafs1/home/dp208/dp208/shared/src/grid-sp4-20221213/Grid/qcd/utils/SUn.h:144:320: error: ‘__T0’ was not declared in this scope; did you mean ‘__y0’?
  144 |   accelerator_for(ss, grid->oSites(), 1, {
      |                                                                                                                                                                                                                                                                                                                                ^
      |                                                                                                                                                                                                                                                                                                                                __y0
/mnt/lustre/tursafs1/home/dp208/dp208/shared/src/grid-sp4-20221213/Grid/qcd/utils/SUn.h: In static member function ‘static void Grid::GaugeGroup<ncolour, group_name>::su2Insert(const Grid::Lattice<Grid::iScalar<Grid::iScalar<Grid::iMatrix<vtype, 2> > > >&, Grid::Lattice<Grid::iScalar<Grid::iScalar<Grid::iMatrix<vtype, ncolour> > > >&, int)’:
/mnt/lustre/tursafs1/home/dp208/dp208/shared/src/grid-sp4-20221213/Grid/qcd/utils/SUn.h:176:273: error: ‘__T1’ was not declared in this scope; did you mean ‘__y1’?
  176 |   accelerator_for(ss, grid->oSites(), 1, {
      |                                                                                                                                                                                                                                                                                 ^
      |                                                                                                                                                                                                                                                                                 __y1
```

--- a/Grid/qcd/utils/GaugeGroup.h
+++ b/Grid/qcd/utils/GaugeGroup.h
@@ -2,7 +2,7 @@
 
 Grid physics library, www.github.com/paboyle/Grid
 
-Source file: ./lib/qcd/utils/SUn.h
+Source file: ./lib/qcd/utils/GaugeGroup.h
 
 Copyright (C) 2015
 
@@ -145,8 +145,8 @@ class GaugeGroup {
   typedef Lattice<vSU2MatrixF> LatticeSU2MatrixF;
   typedef Lattice<vSU2MatrixD> LatticeSU2MatrixD;
 
-#include "Grid/qcd/utils/SUn.h"
-#include "Grid/qcd/utils/Sp2n.h"
+#include "Grid/qcd/utils/SUn.impl"
+#include "Grid/qcd/utils/Sp2n.impl"
 
  public:
   template <class cplx>

--- a/Grid/qcd/utils/GaugeGroup.h
+++ b/Grid/qcd/utils/GaugeGroup.h
@@ -32,15 +32,17 @@ directory
 #ifndef QCD_UTIL_SUN_H
 #define QCD_UTIL_SUN_H
 
+// Important detail: nvcc requires all template parameters to have names.
+// This is the only reason why the second template parameter has a name.
 #define ONLY_IF_SU                                                       \
   typename dummy_name = group_name,                                      \
-           typename = std::enable_if_t <                                 \
+           typename named_dummy = std::enable_if_t <                                 \
                           std::is_same<dummy_name, group_name>::value && \
                       is_su<dummy_name>::value >
 
 #define ONLY_IF_Sp                                                       \
   typename dummy_name = group_name,                                      \
-           typename = std::enable_if_t <                                 \
+           typename named_dummy = std::enable_if_t <                                 \
                           std::is_same<dummy_name, group_name>::value && \
                       is_sp<dummy_name>::value >
 

--- a/Grid/qcd/utils/SUn.impl
+++ b/Grid/qcd/utils/SUn.impl
@@ -5,6 +5,9 @@
 // class GaugeGroup {
 //
 // around it.
+//
+// Please note that the unconventional file extension makes sure that it
+// doesn't get found by the scripts/filelist during bootstrapping.
 
 private:
 static int su2subgroups(GroupName::SU) { return (ncolour * (ncolour - 1)) / 2; }

--- a/Grid/qcd/utils/Sp2n.impl
+++ b/Grid/qcd/utils/Sp2n.impl
@@ -1,3 +1,13 @@
+// This file is #included into the body of the class template definition of
+// GaugeGroup. So, image there to be
+//
+// template <int ncolour, class group_name>
+// class GaugeGroup {
+//
+// around it.
+//
+// Please note that the unconventional file extension makes sure that it
+// doesn't get found by the scripts/filelist during bootstrapping.
 
 private:
 static int su2subgroups(GroupName::Sp) { return (ncolour/2 * (ncolour/2 - 1)) / 2; }

